vignettes/advs.Rmd

```diff
@@ -0,0 +1,407 @@
+vignette: >
+  %\VignetteIndexEntry{Creating a Metabolic ADVS ADaM}
+  %\VignetteEncoding{UTF-8}
+  %\VignetteEngine{knitr::rmarkdown}
+editor_options: 
+  markdown: 
+    wrap: 72
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+
+library(admiraldev)
+```
+
+# Introduction
+
+This article describes creating a vital signs ADaM for metabolic
+clinical trials.
+
+We advise you first consult the `{admiral}` [Creating a BDS Finding ADaM
+vignette](https://pharmaverse.github.io/admiral/articles/bds_finding.html).
+The programming workflow around creating the general set-up of an `ADVS`
+using `{admiral}` functions is the same. We focus on the most common
+endpoints and their derivations mainly found in metabolic trials to
+avoid repeating information and maintaining the same content in two
+places.
+
+## Required Packages
+
+The examples of this vignette require the following packages.
+
+```{r, warning=FALSE, message=FALSE}
+library(admiral)
+library(admiralmetabolic)
+library(pharmaversesdtm)
+library(dplyr, warn.conflicts = FALSE)
+library(lubridate)
+library(stringr)
+```
+
+# Programming Workflow
+
+-   [Read in Data](#readdata)
+-   [Assign `PARAMCD`, `PARAM`, `PARAMN`, `PARCAT1`](#paramcd)
+-   [Derive Additional Parameters (e.g. `BMI` for `ADVS`)](#derive_param)
+-   [Common Metabolic Endpoints](#common_endpoints)
+    -   [Derive Categorization Variables (`AVALCATx`, `BASECATx`)](#cat)
+    -   [Derive Criterion Variables (`CRITy`, `CRITyFL`, `CRITyFLN`)](#crit_vars)
+-   [Remaining ADVS Set-up](#advs_end)
+
+## Read in Data {#readdata}
+
+To start, all data frames needed for the creation of `ADVS` should be
+read into the environment. This will be a company specific process. Some
+of the data frames needed may be `VS` and `ADSL`.
+
+```{r, message=FALSE, warning=FALSE}
+dm_metabolic <- admiralmetabolic::dm_metabolic
+vs_metabolic <- admiralmetabolic::vs_metabolic
+admiral_adsl <- admiral::admiral_adsl
+
+dm <- convert_blanks_to_na(dm_metabolic)
+vs <- convert_blanks_to_na(vs_metabolic)
+admiral_adsl <- convert_blanks_to_na(admiral_adsl)
+```
+
+Within this vignette, the `DM` is used as the basis for `ADSL`:
+
+```{r eval=TRUE}
+# Retrieve required variables from admiral ADSL for this vignette that are not present in DM dataset
+adsl <- dm %>%
+  select(-DOMAIN) %>%
+  mutate(TRT01P = ARM, TRT01A = ACTARM) %>%
+  left_join(admiral_adsl %>% select(USUBJID, TRTSDT, TRTEDT), by = "USUBJID")
+```
+
+The following steps are to merge `ADSL` variables with the source data and
+derive the usual `ADVS` analysis variables. Note that only the sections required for this vignette are covered in the following steps. To get a detailed guidance on all the steps, refer the `{admiral}` [Creating a BDS Finding ADaM vignette](https://pharmaverse.github.io/admiral/articles/bds_finding.html).
+
+Join `ADSL` to your `VS` domain. Only the `ADSL` variables used for derivations are selected at this step.
```

Comment:
From here to the end of the subsection, I would hide these chunks, remove the explanation and accept the vignette is not completely executable as otherwise we are just duplicating what is in the admiral advs vignette.

Reply:
@manciniedoardo  A bit confused here, are we not showing the next step of creating advs as you have marked here ?

Reply:
I just think we should hide the bits where you've just copied from the ADVS vignette (deriving dates, days, merging with adsl)